source/fundamentals/transactions.txt

```diff
@@ -0,0 +1,417 @@
+- ``abortTransaction()``
+
+See the :ref:`Core API example <nodejs-transaction-core-api-example>` for
+a sample transaction implementation.
+
+Callback API
+~~~~~~~~~~~~
+
+The Callback API enables you to run a transaction by passing a callback
+function that encapsulates the series of operations that make up the
+transaction. This API automatically handles certain transaction errors
+returned by the server.
+
+See
+:manual:`TransientTransactionError </core/transactions-in-applications/#std-label-transient-transaction-error>`
+and
+:manual:`UnknownTransactionCommitResult </core/transactions-in-applications/#-unknowntransactioncommitresult->`
+for more information on these errors.
+
+To create a transaction, pass a callback function that encapsulates your
+transaction logic to the ``withTransaction()`` method on a ``Session``
+object. The driver automatically retries the transaction when it encounters
+certain errors reported by the server.
+
+See the :ref:`Callback API example <nodejs-transaction-callback-api-example>`
+for a sample transaction implementation.
+
+.. _nodejs-transaction-settings:
+
+Transaction Settings
+--------------------
+
+When you instantiate a transaction, you can specify the following options to
+set the default behavior for that transaction:
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Setting
+     - Description
+
+   * - ``readConcern``
+     - Specifies how to check for the consistency of the data that the
+       read operations retrieve from replica sets.
+
+       See :manual:`Read Concern </reference/read-concern/>` in the server
+       manual for more information.
+
+   * - ``writeConcern``
+     - Specifies conditions in which to acknowledge the write.
+       See :manual:`Write Concern </reference/write-concern/>` in the
+       server manual for more information.
+
+   * - ``readPreference``
+     - See :manual:`Read Preference </core/read-preference/#read-preference>`
+       in the server manual for more information.
+
+   * - ``maxCommitTimeMS``
+     - Specifies the maximum amount of time to allow a commit action on a
+       transaction to run in milliseconds.
+
+If you do not provide values, the driver uses the client settings.
+
+You can specify the transaction options in the Core API using code that
+resembles the following:
+
+.. code-block:: javascript
+
+   const transactionOptions = {
+     readPreference: 'primary',
+     readConcern: { level: 'local' },
+     writeConcern: { w: 'majority' },
+     maxCommitTimeMS: 1000
+   };
+   session.startTransaction(transactionOptions);
+
+You can specify the transaction options in the Callback API using code that
+resembles the following:
+
+.. code-block:: javascript
+
+   const transactionOptions = {
+     readPreference: 'primary',
+     readConcern: { level: 'local' },
+     writeConcern: { w: 'majority' },
+     maxCommitTimeMS: 1000
+   };
+   await session.withTransaction(myCallbackFunction(session), transactionOptions);
```

Comment:
I don't think this syntax and the explanation below is quite correct, since `myCallbackFunction(session)` implies that you need to pass in the session and the text below could be interpreted to mean that there are other variables that could be defined as parameters of the function. It would be more correct to just have `myCallbackFunction` in this line, and then explain in the lines below that it can use `session` as a parameter (i.e., it will have access to `session` when it is executing). It's also probably important to mention that this function *must* return a promise to work properly.

Reply:
Just to add on a bit, I think this can still be a one liner example like so: 
```suggestion
   await session.withTransaction(async (session) => { /* code here */ }, transactionOptions);
```
And the text below could reference replacing "code here" or however the final formatting ends up.

Reply:
Ok sounds good. I'll use Neal's syntax and explain that 1) you should pass session as a parameter if you want to declare a separate function and 2) it needs to return a Promise.

Reply:
It doesn't look like this feedback has been addressed

Reply:
Sorry, I changed it in one location but not this one. I'll check more carefully in the future!